dppylib/__init__.py

```diff
@@ -132,13 +132,32 @@ def insert_data(db, file_info):
         # Import data
         data_blob = []
         import_collection = db[file_info['collection']]
+        # Check for min and max day in collection
+        min_day = float('inf')
+        max_day = -1
+        min_day_found = import_collection.find_one({'day': {'$exists': True}}, sort=[('day', 1)], projection= {'day': 1})
+        if min_day_found is not None:
+            min_day = min_day_found['day']
+        max_day_found = import_collection.find_one({'day': {'$exists': True}}, sort=[('day', -1)], projection= {'day': 1})
+        if max_day_found is not None:
+            max_day = max_day_found['day']
         for chunk in reader.read_csv(file_info['path']):
             if len(chunk) > 0:
                 if file_info['role'] != 'metadata':
                     chunk_columns = sanitize_columns(chunk.columns.values.tolist())
                     chunk.columns = chunk_columns
                 chunk['path'] = file_info['path']
-                data_blob.extend(chunk.to_dict('records'))
+                if 'day' in chunk: 
+                    chunk_day = int(chunk['day'])
+                    # Add to data blob if there is no existing day range for this collection
+                    if min_day_found is None or max_day_found is None:
+                        data_blob.extend(chunk.to_dict('records'))
+                    # Add to data blob if this day not in existing range
```

Comment:
TYPO: change to `day is not`

Reply:
More shorthand than typo, but happy to change.

Reply:
I agree it is a shorthand but appeared as a grammatical error due to perfection above `there is`

Reply:
Done in d14bd63